### PR TITLE
kernel: kconfig: changing recommendations for SYS_CLOCK_TICKS_PER_SEC

### LINKED ISSUE #25482: outdated recommendations for SYS_CLOCK_TICKS_PER_SEC
The documentation for Kconfig `SYS_CLOCK_TICKS_PER_SEC` assumes that the system clock runs at 1 MHz; other values will not produce the granularity described.

The recommendation should also be more clear that the optimal value is one that when multiplied by a power of two produces the frequency; otherwise the reference to bit-shifting is incorrect.  The specified values are very wrong for Nordic, as they not only require division operations but also accumulate error in the uptime clock.

The recommendations related to tickless may not be entirely accurate, and the suggestion of using 0 to disable timer support should be validated.

### PR DESCRIPTION
**Pull request by yerabolu**
Documentation for kconfig SYS_CLOCK_TICKS_PER_SEC has some outdated
recommendations. Changing them to align with other documentation
under kernel timing.

Fixes: #25482

Signed-off-by: Spoorthy Priya Yerabolu <spoorthy.priya.yerabolu@intel.com>

### CODE REVIEW

File: kernel/Kconfig [hunk 1]
```diff
@@ -502,30 +502,19 @@ config ARCH_HAS_CUSTOM_BUSY_WAIT
 
 config SYS_CLOCK_TICKS_PER_SEC
 	int "System tick frequency (in ticks/second)"
-	default 100 if QEMU_TARGET || SOC_POSIX
```
Comment (nashif) [inline kernel/Kconfig:505]: why are you removing those?

Comment (yerabolu) [reply kernel/Kconfig:505]: sorry, that's a mistake. will update it now

File: include/kernel.h [hunk 2]
```diff
@@ -4607,6 +4607,8 @@ extern int k_mem_slab_init(struct k_mem_slab *slab, void *buffer,
  *
  * This routine allocates a memory block from a memory slab.
  *
+ * @note Can be called by ISRs, but @a timeout must be set to K_NO_WAIT.
```
Comment (nashif) [inline include/kernel.h:4610]: how is this change related to the Kconfig changes in the title of this PR?

**Review by nashif — CHANGES_REQUESTED** [review]

File: include/kernel.h (hunk 2 above)
Comment (yerabolu) [reply include/kernel.h]: They are not related, for some reason my last PR changes got submitted along with this one.  Pushed the changes.

**Review by andyross — APPROVED** [review]

File: kernel/Kconfig [hunk 3]
```diff
@@ -508,24 +508,16 @@ config SYS_CLOCK_TICKS_PER_SEC
...
-	  Other good values are 100, 50, 25, 20 and 10. In this case, some math
-	  is required but is minimized. These are also values that necessitate a
-	  reduced number of clock interrupts per second, at the cost of
-	  granularity (10ms to 100ms).
-
-	  All other values require some extensive 64-bit math, and in some
-	  configurations even require calls to compiler built-in functions, and
-	  can require a non-trivial extra amount of stack space (e.g. around 80
-	  bytes on x86).
+	  For asynchronous timekeeping, the kernel defines a "ticks" concept.  A
```
Comment (nashif) [inline kernel/Kconfig:511]: what used to be easy to read is now condensed in one paragraph. 
Also, "The choice of tick rate is configurable via :c:option:`CONFIG_SYS_CLOCK_TICKS_PER_SEC`" sound like redundant, because this is the same option we are documenting.

File: kernel/Kconfig [hunk 4]
```diff
@@ -508,24 +508,16 @@ config SYS_CLOCK_TICKS_PER_SEC
...
+	  "tick" is the internal count in which the kernel does all its internal
+	  uptime and timeout bookeeping.  Interrupts are expected to be
+	  delivered on tick boundaries to the extent practical, and no
+	  fractional ticks are tracked.  The choice of tick rate is configurable
+	  via :c:option:`CONFIG_SYS_CLOCK_TICKS_PER_SEC`.  Defaults on most
```
Comment (pabigot) [inline kernel/Kconfig:516]: AFAICT we don't ever use RST annotations like `:c:option` in Kconfig help text.

Agree that a paragraph break before "The choice of tick rate" would read better, and the sentence could be "The choice of tick rate is configured by this option.  Defaults on..."  

A sentence like "Also the number of cycles per tick should be chosen so that 1 millisecond is exactly represented by an integral number of ticks." would be worth adding somewhere in that second paragram, since if that is not true `k_uptime_get()` will accumulate error.

Otherwise this is a vast improvement, thanks.

Comment (yerabolu) [reply kernel/Kconfig]: @pabigot : Addressed the changes, could you review this again?

File: kernel/Kconfig (hunk 3 above)
Comment (yerabolu) [reply kernel/Kconfig]: Changed it to a 2 para and removed via :c:option:CONFIG_SYS_CLOCK_TICKS_PER_SEC

**Review by tyrantlizard77 — APPROVED** [review]

**Review by nashif — APPROVED** [review]
